pkg/cloudprovider/aws/object_storage_adapter.go

```diff
@@ -27,14 +27,16 @@ import (
 var _ cloudprovider.ObjectStorageAdapter = &objectStorageAdapter{}
 
 type objectStorageAdapter struct {
-	s3 *s3.S3
+	s3       *s3.S3
+	KMSKeyID string
 }
 
 func (op *objectStorageAdapter) PutObject(bucket string, key string, body io.ReadSeeker) error {
 	req := &s3.PutObjectInput{
-		Bucket: &bucket,
-		Key:    &key,
-		Body:   body,
+		Bucket:      &bucket,
+		Key:         &key,
+		SSEKMSKeyId: &op.KMSKeyID,
```

Comment:
This also needs [`ServerSideEncryption: "aws:kms"`](https://godoc.org/github.com/aws/aws-sdk-go/service/s3#PutObjectInput) if `op.KMSKeyID != ""`.